source/server/configuration_impl.cc

```diff
@@ -127,6 +110,32 @@ void MainImpl::initializeTracers(const envoy::api::v2::Tracing& configuration, I
   }
 }
 
+void MainImpl::initializeStatsSinks(const envoy::api::v2::Bootstrap& bootstrap, Instance& server) {
+  ENVOY_LOG(info, "loading stats sink configuration");
+
+  for (const envoy::api::v2::StatsSink& sink_object : bootstrap.stats_sinks()) {
+    if (sink_object.name().empty()) {
+      throw EnvoyException(
+          "sink object does not have 'name' attribute to look up the implementation");
+    }
+
+    if (!sink_object.has_config()) {
+      throw EnvoyException(
+          "sink object does not contain the 'config' object to configure the implementation");
```

Comment:
I think an empty default config is legit, we could have some filters that don't require config.